packages/backend/src/redirect.ts

```diff
@@ -0,0 +1,24 @@
+type RedirectAdapter = (url: string, response?: any) => void;
+
+type SingUpParams = { signUpUrl: string; url: string; response?: any };
+type SingInParams = { signInUrl: string; url: string; response?: any };
+
+export class RedirectHelper {
+  readonly #redirectAdapter: RedirectAdapter;
+
+  constructor(redirectAdapter: RedirectAdapter) {
+    this.#redirectAdapter = redirectAdapter;
+  }
+
+  toSignup({ signUpUrl, url, response }: SingUpParams): void {
+    this.#redirectAdapter(this.#buildPath(signUpUrl, url), response);
+  }
+
+  toSignIn({ signInUrl, url, response }: SingInParams): void {
+    this.#redirectAdapter(this.#buildPath(signInUrl, url), response);
+  }
+
+  #buildPath(destinationUrl: string, currentUrl: string) {
+    return `${destinationUrl}#/?redirect_url=${currentUrl}`;
```

Comment:
❓ Since we operate on the server context, I expect this to be path-based, not hash-based. Am I missing something?

Reply:
I followed the description of the ticket. Let's take it offline and came back to it with the changes.